include/frg/formatting.hpp

```diff
@@ -12,14 +12,21 @@
 #include <frg/string_stub.hpp>
 
 /* the ranges library is not even partially freestanding for some reason */
-#if __STDC_HOSTED__
+#if __STDC_HOSTED__ && !defined(__clang__)
 #  if __has_include(<ranges>) && __has_include(<algorithm>)
 #    include <ranges>
 #    include <algorithm>
 #    define FRG_HAS_RANGES
 #  endif
 #endif
 
+#if __STDC_HOSTED__
+#	if __has_include(<iostream>)
```

Comment:
Why is this checked here and not in logging.hpp specifically? It's used there and not here.